components/Modal.js

```diff
@@ -29,7 +29,10 @@ export default function Modal({ show = false, setShow, title, children }) {
               leaveFrom="opacity-100 translate-y-0 sm:scale-100"
               leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
             >
-              <Dialog.Panel className="relative transform overflow-hidden rounded-lg bg-white dark:bg-primary-high text-left shadow-xl transition-all m-16 sm:w-full sm:max-w-4xl max-h-screen">
+              <Dialog.Panel
+                className="relative transform overflow-hidden rounded-lg bg-white dark:bg-primary-high text-left shadow-xl transition-all m-16 sm:w-full sm:max-w-4xl max-h-screen overflow-y-auto"
+                style={{ height: "600px" }}

```

Comment:
Please use Tailwind styles rather than custom styles, you can learn more about Tailwind height in the official docs here https://tailwindcss.com/docs/height